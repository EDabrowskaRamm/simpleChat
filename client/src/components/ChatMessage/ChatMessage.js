import React from 'react'

import {
	ChatMessageRoot,
	Header,
	Name,
	Date,
	Content,
	Controls,
	DeleteButton,
	EditButton
} from './ChatMessage.shards'

import { EVENT_TYPE } from '../../utils/consts.js'

const ChatMessage = ({
	name,
	text,
	date,
	isUser,
	removeMessage,
	type
}) => <ChatMessageRoot>
		{(isUser && type !== EVENT_TYPE.MESSAGE_REMOVE) && <Controls>
			<EditButton>...</EditButton>
			<DeleteButton onClick={removeMessage}>
				x
		</DeleteButton>
		</Controls>}
		<Header>
			<Name>{name}</Name> <Date>{date}</Date>
		</Header>
		<Content type={type}>{text}</Content>
	</ChatMessageRoot>

export default ChatMessage