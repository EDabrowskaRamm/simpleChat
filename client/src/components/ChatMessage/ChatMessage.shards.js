import styled from '@emotion/styled'
import { css } from '@emotion/core'

import { EVENT_TYPE } from '../../utils/consts.js'

export const ChatMessageRoot = styled.div`
  position: relative;
  padding-left: 17px;
  padding-bottom: 17px;

  &:first-of-type {
    margin-top: 25px;
  }

  &:hover {
    button {
      display: inline-block;
    }
  }
`

export const Controls = styled.div`
  position: absolute;
  top: -5px;
  right: 0px;
  font-size: 2rem;

  @media (min-width: 768px) {
    top: 3px;
  }
`

const Button = styled.button`
  display: none;
  cursor: pointer;
  border: ${({ theme }) => theme.commons.borders.main};
  border-radius: ${({ theme }) => theme.commons.borderRadius.main};

  &:hover,
  &:focus {
    opacity: .6;
  }
`

export const EditButton = styled(Button)`
  color: ${({ theme }) => theme.colors.text.link};
  padding: 2px 6px;
  margin-right: 10px;
`

export const DeleteButton = styled(Button)`
  color: ${({ theme }) => theme.colors.text.error};
  padding: 2px 8px;
`

export const Header = styled.div`
  display: flex;
  margin: 11px 0 5px;
`

export const Name = styled.p`
  font-weight: bold;
  text-transform: capitalize;
  letter-spacing: 1.5px;
  margin-right: 15px;
  max-width: 280px;

  @media (min-width: 768px) {
    max-width: 435px;
  }
`

export const Date = styled.span`
  color: ${({ theme }) => theme.colors.text.placeholder};
`

export const Content = styled.p`
  letter-spacing: 1.4px;
  line-height: 1.6;

  ${({ type, theme }) => type === EVENT_TYPE.MESSAGE_REMOVE && css`
    color: ${theme.colors.text.placeholder};
    font-style: italic;
  `}
`

export const Edited = styled.p`
  font-size: 1rem;
  font-style: italic;
  color: ${({ theme }) => theme.colors.text.placeholder};
`