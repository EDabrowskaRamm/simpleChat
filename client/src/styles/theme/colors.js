const _white = '#FFF'
const _colorGrey56 = '#868c97'
const _colorGrey61 = '#979c9f'
const _colorGrey63 = '#9a9fa8'
const _colorGrey89 = '#e1e3e5'
const _colorGrey92 = '#ebebeb'

const _pictonBlue = '#3d90f1'
const _oxfordBlue = '#354052'

export const background = {
  primary: _white,
  secondary: _colorGrey92,
}

export const text = {
  primary: _oxfordBlue,
  secondary: _colorGrey63,
  link: _pictonBlue,
  placeholder: _colorGrey61,
  left: _colorGrey56,
}

export const border = {
  input: _colorGrey56,
  chat: _colorGrey89,
  transparent: 'transparent',
}
