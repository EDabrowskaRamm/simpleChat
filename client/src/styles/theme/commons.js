import { border } from './colors'

export const borders = {
  main: `1px solid ${border.chat}`,
  input: `1px solid ${border.input}`,
}
