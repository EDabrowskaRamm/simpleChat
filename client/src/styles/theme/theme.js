import * as colors from './colors'
import * as commons from './commons'

export default {
  colors,
  commons
}
