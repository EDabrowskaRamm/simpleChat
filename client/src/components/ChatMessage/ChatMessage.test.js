import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import ChatMessage from './ChatMessage'

import theme from '../../styles/theme/theme'
import initialState from '../../../__mocks__/fileMock'

expect.addSnapshotSerializer(serializer)

describe('renders ChatMessage', () => {
  it('Snapshot', () => {
    const mockStore = configureStore()
    const mockData = {
      name: initialState.messages.messagesList[0].name,
      text: initialState.messages.messagesList[0].message,
      date: initialState.messages.messagesList[0].date
    }

    const component = renderer.create(
      <Provider store={mockStore(initialState)}>
        <ThemeProvider theme={theme}>
          <ChatMessage name={mockData.name} text={mockData.text} date={mockData.date} />
        </ThemeProvider>
      </Provider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
