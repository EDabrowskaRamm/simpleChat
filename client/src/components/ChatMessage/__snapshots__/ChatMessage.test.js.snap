// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChatMessage Snapshot 1`] = `
.emotion-4 {
  position: relative;
  padding-left: 17px;
  padding-bottom: 17px;
}

.emotion-4:first-of-type {
  margin-top: 25px;
}

.emotion-4:hover button {
  display: inline-block;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 11px 0 5px;
}

.emotion-0 {
  font-weight: bold;
  text-transform: capitalize;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  margin-right: 15px;
}

.emotion-1 {
  color: #979c9f;
}

.emotion-3 {
  -webkit-letter-spacing: 1.4px;
  -moz-letter-spacing: 1.4px;
  -ms-letter-spacing: 1.4px;
  letter-spacing: 1.4px;
  line-height: 1.6;
}

<div
  className="emotion-4"
>
  <div
    className="emotion-2"
  >
    <p
      className="emotion-0"
    >
      Jane Downey
    </p>
     
    <span
      className="emotion-1"
    >
      12:58:24 
    </span>
  </div>
  <p
    className="emotion-3"
    type="msgevent"
  >
    After losing his wife seven years earlier, the eccentric Dr. John Dolittle, famed doctor and veterinarian of Queen Victoriaâ€™s England, hermits himself away behind the high walls of Dolittle Manor with only his menagerie of exotic animals for company.
  </p>
</div>
`;
